e2e/getenvoy_run_test.go

```diff
@@ -30,113 +33,149 @@ import (
 	"github.com/tetratelabs/getenvoy/internal/test/morerequire"
 )
 
-const shutdownTimeout = 2 * time.Minute
+var (
+	// staticFilesystemConfig shows Envoy reading a file referenced from the CWD
+	//go:embed static-filesystem.yaml
+	staticFilesystemConfig  []byte
+	adminAddressPathPattern = regexp.MustCompile(`--admin-address-path ([^ ]+)`)
+	envoyStartedLine        = "starting main dispatch loop"
+)
 
 // TestGetEnvoyRun runs the equivalent of "getenvoy run"
 //
 // See TestMain for general notes on about the test runtime.
 func TestGetEnvoyRun(t *testing.T) {
 	t.Parallel() // uses random ports so safe to run parallel
 
-	// Below is the minimal config needed to run envoy 1.18+, non-windows <1.18 need access_log_path: '/dev/stdout'
-	c := getEnvoy(`run`, "--config-yaml", "admin: {address: {socket_address: {address: '127.0.0.1', port_value: 0}}}")
-
-	stdout, stderr, shutdown := c.start(t, shutdownTimeout)
-
-	// The underlying call is conditional to ensure errors that raise before we stop the server, stop it.
-	deferredShutdown := shutdown
-	defer func() {
-		if deferredShutdown != nil {
-			deferredShutdown()
-		}
-	}()
-
-	runDir := requireRunDir(t, stdout, c)
-	requireEnvoyReady(t, runDir, stderr, c)
+	// Below is the minimal config needed to run Envoy 1.18+, non-windows <1.18 need access_log_path: '/dev/stdout'
+	args := []string{"run", "--config-yaml", "admin: {address: {socket_address: {address: '127.0.0.1', port_value: 0}}}"}
 
-	log.Printf(`shutting down Envoy after running [%v]`, c)
-	shutdown()
-	deferredShutdown = nil
+	c, runArchive := envoyRunTest(t, nil, args...)
```

Comment:
@mathetake sorry this is where nil is.. basically we aren't testing anything about the process before it is shutdown